toolchain/sem_ir/inst_fingerprinter.cpp

```diff
@@ -34,11 +35,39 @@ struct Worklist {
   llvm::SmallVector<llvm::stable_hash> contents = {};
   // Known cached instruction fingerprints. Each item in `todo` will be added to
   // the cache if not already present.
-  Map<std::pair<const File*, InstId>, uint64_t>* fingerprints;
+  llvm::SmallVector<
+      std::pair<const File*, FixedSizeValueStore<InstId, uint64_t>>>*
+      fingerprints;
 
   // Finish fingerprinting and compute the fingerprint.
   auto Finish() -> uint64_t { return llvm::stable_hash_combine(contents); }
 
+  // Gets the known fingerprint from the cache, or returns 0.
+  auto GetFingerprint(const File* file, InstId inst_id) -> uint64_t {
+    for (auto& [f, values] : *fingerprints) {
```

Comment:
Isn't this going to scale poorly with # of files?

Reply:
Can we index `fingerprints` using the file's `CheckIRId` instead?

Reply:
I guess I only need to do that if I want to use a FixedSizeValueStore<CheckIRId, ...> but to do that I would then also need to construct each inner `FixedSizeValueStore<InstId, uint64_t>` and I don't have all the `File*` to do that. So I guess I can lazily construct those.

Reply:
Ok I moved the `total_ir_count` from `Check::Context` to `SemIR::File` so we can use it everywhere and used a FixedSizeValueStore so I don't have to work with the `.index` of the `CheckIRId` which always feels wrong.